mirdata/datasets/compmusic_otmm_makam.py

```diff
@@ -0,0 +1,249 @@
+
+    Attributes:
+        makam (str): string referring to the makam represented in the track
+        tonic (float): tonic annotation
+        mbid (str): MusicBrainz ID of the track
+
+    Cached Properties:
+        pitch (F0Data): pitch annotation
+        mb_tags (dict): dictionary containing the raw editorial track metadata from music brainz
+
+    """
+
+    def __init__(
+        self,
+        track_id,
+        data_home,
+        dataset_name,
+        index,
+        metadata,
+    ):
+        super().__init__(
+            track_id,
+            data_home,
+            dataset_name,
+            index,
+            metadata,
+        )
+        # Annotation paths
+        self.pitch_path = core.none_path_join(
+            [self._data_home, self._track_paths["pitch"][0]]
+        )
+        self.mb_tags_path = core.none_path_join(
+            [self._data_home, self._track_paths["metadata"][0]]
+        )
+
+        # Get attributes
+        self.tonic = self._track_metadata.get("tonic")
+        self.makam = self._track_metadata.get("makam")
+        self.mbid = self._track_metadata.get("mbid")
+
+    @core.cached_property
+    def pitch(self):
+        return load_pitch(self.pitch_path)
+
+    @core.cached_property
+    def mb_tags(self):
+        return load_mb_tags(self.mb_tags_path)
+
+    def to_jams(self):
+        """Get the track's data in jams format
+
+        Returns:
+            jams.JAMS: the track's data in jams format
+
+        """
+        return jams_utils.jams_converter(
+            f0_data=[(self.pitch, "pitch")],
+            metadata={
+                "makam": self.makam,
+                "tonic": self.tonic,
+                "mbid": self.mbid,
+                "duration": self.mb_tags["duration"],
+                "metadata": self.mb_tags,
+            },
+        )
+
+
+@io.coerce_to_string_io
+def load_pitch(fhandle: TextIO) -> annotations.F0Data:
+    """Load pitch
+
+    Args:
+        pitch path (str): Local path where the pitch annotation is stored.
+            If `None`, returns None.
+
+    Returns:
+        F0Data: pitch annotation
+
+    """
+    time_step = 0.0029
+
+    times = []
+    freqs = []
+    reader = csv.reader(fhandle, delimiter=",")
+    for line in reader:
+        freqs.append(float(line[0]))
+
+    for i in np.arange(len(freqs)):
+        times.append(round(float(time_step * i), 4))
+
+    times = np.array(times)
+    freqs = np.array(freqs)
+    confidence = (freqs > 0).astype(float)
+    return annotations.F0Data(times, freqs, confidence)
+
+
+@io.coerce_to_string_io
+def load_mb_tags(fhandle: TextIO) -> dict:
+    """Load track metadata
+
+    Args:
+        track metadata path (str): Local path where the metadata of the track is stored.
+            If `None`, returns None.
+
+    Returns:
+        Dict: metadata of the track
+
+    """
+    mb_tags = json.load(fhandle)
+
+    return mb_tags
```

Comment:
nit - this function can simply `return json.load(fhandle)`!

Reply:
True! Fixed now:)